muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -419,8 +421,11 @@ public Response searchStudents(
             logger.severe(String.format("Student %s in search index not found in Muikku", studentIdentifier));
             continue;
           }
-          String emailAddress = userEmailEntityController.getUserDefaultEmailAddress(userEntity, true);
-
+          String emailAddress = "";
+          User student = userController.findUserByUserEntityDefaults(userEntity);
```

Comment:
This finds the student that has currently been marked as active in Pyramus. It might not be the same as the student returned from the search. Please use...

userController.findUserByIdentifier(studentIdentifier);

...instead.